### ISSUE TITLE
Export inference deviates from PyTorch: extra detections and shifted keypoints

### ISSUE THREAD

**Issue by tom21100227** [talmolab/sleap-nn#466]
## Summary

The ONNX/TensorRT export inference path (`sleap-nn track`) produces **more instances** and **deviated keypoints** compared to the native PyTorch inference path (`sleap-nn predict`). This issue documents the three highest-impact root causes identified through systematic comparison, along with a proposed fix plan.

### 1. NMS algorithm mismatch (primary cause of extra detections)

**Export path** (`sleap_nn/export/wrappers/base.py:47-48`):
```python
pooled = F.max_pool2d(confmaps, kernel_size=3, stride=1, padding=1)
is_peak = (confmaps == pooled) & (confmaps > 0)
```

**PyTorch path** (`sleap_nn/inference/peak_finding.py:306-320`):
```python
kernel = torch.tensor([[1, 1, 1], [1, 0, 1], [1, 1, 1]])  # center-excluded
max_img = morphological_dilation(flat_img, kernel)
argmax_and_thresh_img = (cms > max_img) & (cms > threshold)
```

The key differences:
- **Center inclusion**: `max_pool2d` includes the center pixel. The PyTorch kernel explicitly excludes it (`[1,0,1]` center row). This means `confmaps == pooled` is true for **any pixel on a plateau**, not just strict local maxima.
- **Equality vs strict inequality**: Export uses `==` (matches plateaus), PyTorch uses `>` (only strict maxima). Flat regions in confidence maps produce many spurious peaks with the export NMS.

### 2. Missing peak threshold in export wrappers

The export wrappers hardcode `confmaps > 0` as the threshold:
```python
is_peak = (confmaps == pooled) & (confmaps > 0)  # base.py:48
valid = values > 0                                  # base.py:60
```

The PyTorch path uses a configurable threshold (default 0.2):
```python
argmax_and_thresh_img = (cms > max_img) & (cms > threshold)
```

This means the export path accepts very low-confidence noise as valid peaks, producing additional spurious detections that the PyTorch path correctly filters out.

The `peak_conf_threshold` parameter exists in the `predict` CLI command but is only applied **after** the ONNX graph runs (during CPU-side post-processing for bottom-up). The NMS inside the ONNX graph itself has no threshold, so it wastes top-K slots on noise peaks.

### 3. TF32 precision in TensorRT FP32 mode

TensorRT enables TF32 (TensorFloat-32) by default, even when `precision="fp32"` is requested. TF32 uses 10-bit mantissa vs 23-bit for IEEE FP32, which can introduce small numerical differences in confidence map values. This is a minor contributor but affects reproducibility.

**Current code** (`sleap_nn/export/exporters/tensorrt_exporter.py:180-187`):
```python
if precision == "fp16":
    if builder.platform_has_fast_fp16:
        config.set_flag(trt.BuilderFlag.FP16)
# No explicit handling for fp32 — TF32 remains enabled by default
```

### Fix 1: Replace max_pool2d NMS with center-excluded neighbor max

Replace the `max_pool2d` + equality check with a neighbor-max approach using directional shifts (all ONNX-safe ops):

```python
@staticmethod
def _neighbor_max(x: torch.Tensor) -> torch.Tensor:
    """Max of 8 neighbors, excluding center pixel."""
    p = F.pad(x, [1, 1, 1, 1])  # pad with 0
    return torch.max(torch.max(torch.max(torch.max(
        torch.max(torch.max(torch.max(
            p[:, :, :-2, :-2],   # top-left
            p[:, :, :-2, 1:-1]), # top
            p[:, :, :-2, 2:]),   # top-right
            p[:, :, 1:-1, :-2]), # left
            p[:, :, 1:-1, 2:]),  # right
            p[:, :, 2:, :-2]),   # bottom-left
            p[:, :, 2:, 1:-1]),  # bottom
            p[:, :, 2:, 2:])     # bottom-right

# Then use strict inequality:
is_peak = (confmaps > neighbor_max) & (confmaps > threshold)
```

### Fix 2: Add configurable `peak_threshold` to all wrapper constructors

Thread a `peak_threshold` parameter (default `0.2`) through all wrapper `__init__` methods and pass it to `_find_topk_peaks()`, `_find_topk_peaks_per_node()`, and `_find_global_peaks()`. The threshold gets baked into the ONNX graph at export time.

**Affected files:**
- `sleap_nn/export/wrappers/base.py` — add threshold param to all three peak-finding methods
- `sleap_nn/export/wrappers/centroid.py`
- `sleap_nn/export/wrappers/topdown.py`
- `sleap_nn/export/wrappers/bottomup.py`
- `sleap_nn/export/wrappers/single_instance.py`
- `sleap_nn/export/wrappers/centered_instance.py`
- `sleap_nn/export/wrappers/topdown_multiclass.py`
- `sleap_nn/export/wrappers/bottomup_multiclass.py`
- `sleap_nn/export/cli.py` — thread threshold through wrapper construction

### Fix 3: Disable TF32 for strict FP32 TensorRT exports

Add explicit `config.clear_flag(trt.BuilderFlag.TF32)` when `precision == "fp32"` in both `_export_tensorrt_onnx()` and `_export_tensorrt_jit()`.

## Out of Scope (known minor differences)

These are intentionally deferred as they have minimal impact:

1. **Integral regression refinement** — Would require significant ONNX graph changes. Documented as known ~0.5-2px deviation when enabled.
2. **Crop extraction method** (`grid_sample` vs `unfold`) — Functionally different but both valid. Sub-pixel difference.
3. **PAF line sampling** (bilinear vs nearest-neighbor) — The bilinear approach in export is arguably more accurate.
4. **Anti-aliased resize** — Only affects `input_scale != 1.0` cases.

## Verification

The existing accuracy tests in `tests/export/test_export_accuracy.py` compare PyTorch vs ONNX on the `minimal_instance_bottomup` checkpoint. Current baselines show 0.0 px deviation when both paths use matching settings — these tests will verify the fix doesn't regress.

## Impact

Fixes 1 and 2 together should eliminate the extra spurious detections and bring instance counts in line with the PyTorch path. Fix 3 improves FP32 TensorRT reproducibility.
